saul-core/src/test/scala/joinTraining/intializeSparseNetwork.scala

```diff
@@ -0,0 +1,94 @@
+package joinTraining
```

Comment:
looking back to this test, this folder is meant to keep the tests related to `structured output prediction/join training` so I put the initialization test there. I think it can be added if the foldr name looks ok, otherwise let me know where you want to move it.